coq/coq-system.el

```diff
@@ -432,7 +432,8 @@ LOADPATH, CURRENT-DIRECTORY, PRE-V85: see `coq-include-options'."
   "Build a list of options for coqc.
 LOADPATH, CURRENT-DIRECTORY, PRE-V85: see `coq-coqc-prog-args'."
   (append
-   (if (coq--supports-topfile) (cons "-topfile" (cons buffer-file-name nil)) "")
+   (if (and (coq--supports-topfile) buffer-file-name)
+       (cons "-topfile" (cons buffer-file-name nil)) "")
```

Comment:
Btw (not from this pr) this cons cons thing can be written `(list "-topfile" buffer-file-name)`